dandi/dandiarchive.py

```diff
@@ -142,15 +142,27 @@ def navigate(
         If ``strict`` is true, then `get_dandiset()` is called with
         ``lazy=False`` and `get_assets()` is called with ``strict=True``; if
         ``strict`` is false, the opposite occurs.
+
+        If ``authenticate`` is true, then
+        `~dandi.dandiapi.DandiAPIClient.dandi_authenticate()` will be called on
+        the client before returning it.  If it is `None` (the default), the
+        method will only be called if the resource(s) the URL refers to are
+        embargoed.
```

Comment:
```suggestion
        method will only be called if the request to URL requires authentication
        (e.g., if the resource(s) are embargoed).
```
just to make it a bit closer to the handling, and for future proofing (e.g., if Private/Restricted access model is added).